packages/javascript/bh-shared-ui/src/components/HelpTexts/CodeController/CodeController.tsx

```diff
@@ -0,0 +1,155 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { Button, Typography } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { PropsWithChildren, useRef, useState } from 'react';
+import clsx from 'clsx';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faAlignJustify, faCopy } from '@fortawesome/free-solid-svg-icons';
+import { copyToClipboard } from '../../../utils/copyToClipboard';
+
+export const useStyles = makeStyles((theme) => ({
+    codeController: {
+        position: 'relative',
+        '& .code': {
+            'text-wrap': 'nowrap',
+            overflow: 'scroll',
+        },
+        '& .wrapped': {
+            'text-wrap': 'wrap',
+        },
+        '& .scrollLeft': {
+            'box-shadow': 'inset -5px 0px 5px black;',
+        },
+        '& .scrollRight': {
+            'box-shadow': 'inset 5px 0px 5px black;',
+        },
+        '& .scrollRight.scrollLeft': {
+            'box-shadow': 'inset 5px 0px 5px 0px black, inset -5px 0px 5px 0px black',
+        },
+        '& .codeController': {
+            display: 'flex',
+            justifyContent: 'flex-end',
+            position: 'absolute',
+            right: '0',
+            borderBottom: '1px solid white',
+            width: '100%',
+
+            '& button': {
+                color: 'white',
+                transition: 'opacity 100ms ease-in-out',
+            },
+        },
+    },
+}));
+
+interface Props {
+    hideCopy?: boolean;
+    hideWrap?: boolean;
+}
+
+/**
+ * Wraps <Typography component="pre"> to add controls for copy/paste and code wrapping.
+ * Implementation: please wrap code block in {`<coding>`}
+ * @param hideCopy - default true - display copy text button
+ * @param hideWrap - default true - display wrap text button
+ * @returns
+ */
+function CodeController(props: PropsWithChildren<Props>) {
+    const { hideCopy = false, hideWrap = false, children } = props;
```

Comment:
Should either the comment or value setting be updated so that they are the same?

Reply:
ahh I updated the component and forgot to update the docs, thanks for calling this out!